pkg/codegen/java.go

```diff
@@ -47,7 +48,19 @@ func GenerateJava(pg *PackageGenerator, name string) (map[string]*bytes.Buffer,
 
 	langName := "java"
 	oldName := pkg.Name
-	pkg.Name = name
+	pkg.Name = cs.PackageName
+	pkg.Namespace = cs.PackageNamespace
+
+	// Override Java base package to avoid pulumi-java defaulting logic adding an extra "com." prefix
+	if pkg.Language == nil {
+		pkg.Language = map[string]interface{}{}
+	}
+	pkg.Language[langName] = javaGen.PackageInfo{BasePackage: cs.PackageNamespace}
```

Comment:
For symmetry, should you set `com.pulumi` by default if `cs.PackageNamespace` is empty?
```java
pkg.Language[langName] = javaGen.PackageInfo{BasePackage: "com.pulumi"}
```

Reply:
(nit) I would still advocate for this tool having a built-in default of `com.pulumi`, regardless of the underlying behavior.  I think it is a simpler and more consistent approach.